src/preact/base-element.js

```diff
@@ -142,6 +142,15 @@ const childIdGenerator = sequentialIdGenerator();
  * @template API_TYPE
  */
 export class PreactBaseElement extends AMP.BaseElement {
+  /**
+   * @return {boolean}
+   * @nocollapse
+   */
+  static reqiuresShadowDom() {
+    // eslint-disable-next-line local/no-static-this
+    return usesShadowDom(this);
```

Comment:
I'd way prefer for this method that you manually define the method as static, es5 style. In its current usage, I think it'll actually work as intended, even in a minified dist build.

As long as it is called via `ClassName.requiresShadowDom()`.
What will **silently break** is the moment a dev calls `instance.requiresShadowDom()`.

Reply:
What? `instance.requiresShadowDom` is undefined, so calling it will throw.

Reply:
You're right. I misrepresented [the issue](https://github.com/ampproject/amphtml/issues/29499#issuecomment-668208427), its slightly different (but I think still makes me uncomfortable because its so complicated and a ticking time bomb)

Reply:
To clarify: all good here. No changes are needed?

Reply:
No, I think this is fine.